controllers/rcj_soccer_referee_supervisor/referee/supervisor.py

```diff
@@ -111,60 +118,101 @@ def _update_positions(self):
             # TODO: update self.robot_in_penalty_counter if the robot
             #       is located in penalty area
 
-    def draw_scores(self, blue: int, yellow: int):
+    def add_event_subscriber(self, subscriber: EventHandler):
+        """Add new event subscriber.
+
+        Args:
+            subscriber (EventHandler): Instance inheriting EventHandler
         """
-        Visualize (draw) the provide scores for both the blue and the yellow
-        teams.
+        self.eventer.subscribe(subscriber)
+
+    def draw_scores(self, blue: int, yellow: int):
+        """Visualize (draw) the provide scores for both the blue and
+        the yellow teams.
 
         Args:
             blue (int): score of the blue team
             yellow (int): score of the yellow team
         """
 
-        self.setLabel(0, str(blue),
-                      # X and Y positions
-                      0.92, 0.01,
-                      # Size and color
-                      0.1, 0x0000ff,
-                      # Transparency and Font
-                      0.0, "Tahoma")
-
-        self.setLabel(1, str(yellow),
-                      # X and Y positions
-                      0.05, 0.01,
-                      # Size and color
-                      0.1, 0xffff00,
-                      # Transparency and Font
-                      0.0, "Tahoma")
-
-    def draw_time(self, time):
-        """
-        Visualize (draw) the current match time
+        self.setLabel(
+            LabelIDs.BLUE_SCORE.value,
+            str(blue),
+            0.92,  # X position
+            0.01,  # Y position
+            0.1,  # Size
+            0x0000ff,  # Color
+            0.0,  # Transparency
+            "Tahoma",  # Font
+        )
+
+        self.setLabel(
+            LabelIDs.YELLOW_SCORE.value,
+            str(yellow),
+            0.05,  # X position
+            0.01,  # Y position
+            0.1,  # Size
+            0xffff00,  # Color
+            0.0,  # Transparency
+            "Tahoma"  # Font
+        )
+
+    def draw_time(self, time: int):
+        """Visualize (draw) the current match time
 
         Args:
             time (int): the current match time
         """
 
-        time_str = "%02d:%02d" % (time // 60, time % 60)
-        self.setLabel(2, time_str, 0.45, 0.01, 0.1, 0x000000, 0.0, "Arial")
+        self.setLabel(
+            LabelIDs.TIME.value,
+            time_to_string(time),
+            0.45,
+            0.01,
+            0.1,
+            0x000000,
+            0.0,
+            "Arial",
+        )
+
+    def draw_event_messages(self):
+        """Visualize (draw) the event messages from queue"""
+        messages = []
+        for time, msg in self.event_messages_to_draw:
+            messages.append("{} - {}".format(time_to_string(time), msg))
+
+        if messages:
```

Comment:
@Adman it seems like currently the messages are listed here basically forever but I guess we'd want something on the order of "message disappears after X seconds", where X is configurable. That can certainly wait for another PR though :)

Reply:
Yes, let's check and discuss tonight what would look nicer and fix it in another PR if needed.